### DISCUSSION TITLE
sonar scanned the repo and reported  536 bugs and 43 security hotspots

### DISCUSSION THREAD

**Discussion by tuhao1020** [open-telemetry/opentelemetry-java-instrumentation#6078] [Q&A] [answered]
<img width="942" alt="image" src="https://user-images.githubusercontent.com/26807520/169838332-65679a6b-260b-420e-a18f-94e95680a426.png">

Has the community ever done a static scan of this repo?

**Comment by trask** [comment, marked as answer]
hi @tuhao1020! we use errorprone as our static analysis tool (built into the build) and CodeQL as our security scanning tool (running daily)

**Reply by tuhao1020** [reply]
Is the difference in scanning rules causing the different results? 
<img width="1304" alt="image" src="https://user-images.githubusercontent.com/26807520/170160506-440b5321-7b59-4efb-bb3d-6a19ad03c75f.png">
<img width="970" alt="image" src="https://user-images.githubusercontent.com/26807520/170160677-cd9fdb5a-4329-40be-82f2-a21ceb0fb023.png">

**Reply by anuraaga** [reply]
Hi @tuhao1020 - yes it is a different in the rules and coverage of scanning. It looks like your tool is configured to check for style-related issues such as reusing parameter variables - I would generally recommend not running tools with style-related checks on external repositories as these are not actually bugs but often personal preference. For these in particular, the only way to use byte-buddy (`Advice.Local` in that snippet) is to use this pattern so it wouldn't be possible to change anyways.

errorprone is our primary checker, which is run on every compilation. You can check out the config here

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/conventions/src/main/kotlin/otel.errorprone-conventions.gradle.kts#L17

**Reply by tuhao1020** [reply]
Thanks!
